dask/dataframe/backends.py

```diff
@@ -404,7 +403,7 @@ def _nonempty_series(s, idx=None):
     elif is_datetime64tz_dtype(dtype):
         entry = pd.Timestamp("1970-01-01", tz=dtype.tz)
         data = [entry, entry]
-    elif is_categorical_dtype(dtype):
+    elif isinstance(dtype, pd.CategoricalDtype):
```

Comment:
It's faster, by a lot. Doesn't matter if you are calling it only once, but since it's not more complex I wouldn't use ``is_dtype_equal`` if possible.  (can probably count the occasions I used it on one hand)

```
dtype = pd.CategoricalDtype()
%timeit isinstance(dtype, pd.CategoricalDtype)
48.5 ns ± 0.163 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)
%timeit pd.api.types.is_dtype_equal(dtype, pd.CategoricalDtype())
1.44 µs ± 3.67 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)
```